src/test/java/dk/cngroup/trainings/spring/springassignment/service/UserServiceTest.java

```diff
@@ -0,0 +1,84 @@
+package dk.cngroup.trainings.spring.springassignment.service;
+
+import dk.cngroup.trainings.spring.springassignment.exception.user.UserEmailExistsException;
+import dk.cngroup.trainings.spring.springassignment.model.User;
+import dk.cngroup.trainings.spring.springassignment.repository.UserRepository;
+import dk.cngroup.trainings.spring.springassignment.service.user.UserService;
+import dk.cngroup.trainings.spring.springassignment.service.user.UserServiceImpl;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+import java.util.Optional;
+
+public class UserServiceTest {
+
+	private UserService userService;
+	private UserRepository userRepository;
+	private User walid;
+
+	@Before
+	public void before() {
+		userRepository = Mockito.mock(UserRepository.class);
+		userService = new UserServiceImpl(userRepository);
+		walid = new User(1L, "Walid", "Cheikh", "walid@email.com",
+				"0123");
+	}
+
+	@Test
+	public void addValidUserTest() throws UserEmailExistsException {
+		// Happy path:
+
+		User walidInDb = new User(1L, "Walid", "Cheikh", "walid@email.com",
+				new BCryptPasswordEncoder().encode("1230"));
+		Mockito.when(userRepository.existsByEmail(walid.getEmail()))
+				.thenReturn(false);
+		Mockito.when(userRepository.save(walid))
+				.thenReturn(walidInDb);
+
+		User user = userService.addUser(walid);
+
+		Assert.assertNotNull(user);
+		Assert.assertEquals(user.getEmail(), walid.getEmail());
```

Comment:
maybe also Mockito.verify call would be useful here, to make sure it was saved

Reply:
I will add it.